lib/launcher.rb

```diff
@@ -127,9 +127,8 @@ def launch
 
       (t1 || t2).join
     rescue SignalException => e
-      # This is to prevent the exception handler below from catching SignalException exceptions.
       logger.info("Caught #{e}. Exiting")
-      raise
+      e.signo == 15 ? exit(143) : raise # trap SIGTERM, used by systemd to stop/restart the service
```

Comment:
Why 143? Time to create a table of exit codes at one place to get things in order?

Looks good.